arithmetization/src/test/java/net/consensys/linea/zktracer/forkSpecific/systemTransaction/SystemTransactionTests.java

```diff
@@ -0,0 +1,141 @@
+                testInfo, BEACONROOT_ADDRESS, DEFAULT_TIME_STAMP))
+        .run(testInfo);
+  }
+
+  private static Stream<Arguments> scenariiForSystemContract() {
+    final List<Address> supportedSystemContracts =
+        List.of(BEACONROOT_ADDRESS, HISTORY_STORAGE_ADDRESS);
+    final List<Arguments> scenarii = new ArrayList<>();
+    for (Address systemContract : supportedSystemContracts) {
+      for (int systemContractDeployedBeforeBlockNumber = 0;
+          systemContractDeployedBeforeBlockNumber <= 2;
+          systemContractDeployedBeforeBlockNumber++) {
+        scenarii.add(Arguments.of(systemContract, systemContractDeployedBeforeBlockNumber));
+      }
+    }
+    return scenarii.stream();
+  }
+
+  @ParameterizedTest
+  @MethodSource("scenariiForSystemContract")
+  void genesisBlockTest(Address systemContract, int systemContractDeployedBeforeBlockNumber) {
+
+    // random sender account that will send a tx to system account
+    final KeyPair senderKeyPair = new SECP256K1().generateKeyPair();
+    final Address senderAddress =
+        Address.extract(Hash.hash(senderKeyPair.getPublicKey().getEncodedBytes()));
+    final ToyAccount senderAccount =
+        ToyAccount.builder().balance(Wei.fromEth(123)).nonce(0).address(senderAddress).build();
+
+    final ToyAccount deployerOf2935 = null;
+    final Transaction deploy2935 =
+            ToyTransaction.builder()
+                    .sender(deployerOf2935)
+                    // .keyPair()
+                    .value(Wei.of(0))
+                    .build();
+    final ToyAccount deployerOf4788 = null;
+    final Transaction deploy4788 =
+            ToyTransaction.builder()
+                    .sender(deployerOf4788)
+                    // .keyPair()
+                    .value(Wei.of(0))
+                    .build();
+
+    final ToyAccount callerOf2935 =
+        ToyAccount.builder()
+            .address(Address.wrap(leftPadTo(Bytes.fromHexString("0x2935"), Address.SIZE)))
+            .code(
+                    byteCodeCallingBeaconRootSystemAccountFromCallData(
+                    testInfo, HISTORY_STORAGE_ADDRESS))
+            .build();
+    final ToyAccount callerOf4788 =
+        ToyAccount.builder()
+            .address(Address.wrap(leftPadTo(Bytes.fromHexString("0x4788"), Address.SIZE)))
+            .code(
+                    byteCodeCallingBeaconRootSystemAccountFromCallData(
+                    testInfo, BEACONROOT_ADDRESS))
+            .build();
+
+    final List<Transaction> genesisBlockTransactions = new ArrayList<>();
+    final List<Transaction> blockNb1Transactions = new ArrayList<>();
+    final List<Transaction> blockNb2Transactions = new ArrayList<>();
+
+    final MultiBlockExecutionEnvironment.MultiBlockExecutionEnvironmentBuilder builder =
+        MultiBlockExecutionEnvironment.builder(testInfo)
+            .startingBlockNumber(0)
+            .systemContractDeployedPriorToConflation(systemContractDeployedBeforeBlockNumber == 0);
```

Comment:
I'm not understanding: the loop allows values 0, 1, 2 but the above treats cases 1 and 2 the same. What is the purpose of the 3 cases ?